src/encoded/static/components/item-pages/HiGlassViewConfigView.js

```diff
@@ -153,32 +160,156 @@ export class HiGlassViewConfigTabView extends React.PureComponent {
                         'style' : 'danger'
                     });
                     this.setState({ 'saveLoading' : false });
-                }, 
+                },
+                // We're only updating this object's view conf.
                 JSON.stringify({ 'viewconfig' : currentViewConf })
             );
         });
     }
 
+    /**
+    * Create a new higlass viewconfig for the user, based on the current data.
+    * @returns Nothing
+    */
     handleSaveAs(evt){
         evt.preventDefault();
-        // TODO
+
+        var hgc                 = this.getHiGlassComponent(),
+            currentViewConfStr  = hgc && hgc.api.exportAsViewConfString(),
+            currentViewConf     = currentViewConfStr && JSON.parse(currentViewConfStr);
+
+        if (!currentViewConf){
+            throw new Error('Could not get current view configuration.');
+        }
+
+        // Make sure you are logged in
+        if (this.props.session !== true) {
+            Alerts.queue({
+                'title' : "Please log in.",
+                'message' : "Please log in or sign in before saving this display.",
+                'style' : 'danger'
+            });
+        }
+
+        // Generate a new title and description based on the current display.
+        var userDetails = JWT.getUserDetails();
+        let userFirstName = "unknown";
+        if (userDetails && typeof userDetails.first_name === 'string' && userDetails.first_name.length > 0) userFirstName = userDetails.first_name;
+
+        const viewConfTitle = this.props.context.title + "-" + userFirstName + "'s copy";
```

Comment:
This will return` HiGlass Display Name-Bob's copy`. Maybe the `"-"` should instead be `" - "`?